launch_testing/test/launch_testing/test_resolve_process.py

```diff
@@ -66,3 +71,120 @@ def test_unlaunched_process_lookup_with_substitutions(self):
         # resolved by the launch system, we won't be able to take a guess at the command
         self.assertIn('Unknown', str(cm.exception))
+
+
+class TestStringProcessResolution(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # Run a launch so we can get some real looking proc_info and proc_output objects to test
+        # against
+        proc_command = os.path.join(
+            ament_index_python.get_package_prefix('launch_testing'),
+            'lib/launch_testing',
+            'good_proc',
+        )
+        proc_env = os.environ.copy()
+        proc_env['PYTHONUNBUFFERED'] = '1'
+
+        def generate_test_description(ready_fn):
+            no_arg_proc = launch.actions.ExecuteProcess(
+                cmd=[proc_command],
+                env=proc_env
+            )
+
+            one_arg_proc = launch.actions.ExecuteProcess(
+                cmd=[proc_command, '--one-arg'],
+                env=proc_env
+            )
+
+            two_arg_proc = launch.actions.ExecuteProcess(
+                cmd=[proc_command, '--two-arg', 'arg_two'],
+                env=proc_env
+            )
+
+            ld = launch.LaunchDescription([
+                no_arg_proc,
+                one_arg_proc,
+                two_arg_proc,
+                launch.actions.OpaqueFunction(function=lambda ctx: ready_fn())
+            ])
+
+            return (ld, locals())
+
+        arr = []
+
+        class PreShutdownTests(unittest.TestCase):
+
+            def test_wait_self(self,
+                               proc_output,
+                               proc_info,
+                               no_arg_proc,
+                               one_arg_proc,
+                               two_arg_proc):
+                proc_output.assertWaitFor('--one-arg')
+                proc_output.assertWaitFor('--two-arg')
+                proc_output.assertWaitFor('arg_two')
+
+                arr.append(proc_info)
+
+        # Set up a fake module containing the test data:
+        test_module = types.ModuleType('test_module')
+        test_module.generate_test_description = generate_test_description
+        test_module.FakePreShutdownTests = PreShutdownTests
+
+        # Run the test:
+        runner = LaunchTestRunner(
+            LoadTestsFromPythonModule(test_module)
+        )
+
+        runner.run()
+
+        # Grab the very realistic proc_info object to use for tests below
+        cls.proc_info = arr[0]
+
+    def test_proc_string_lookup_multiple_args(self):
+        found_proc = launch_testing.util.resolveProcesses(
+            self.proc_info,
+            process='good_proc',
+            cmd_args=['--two-arg', 'arg_two']
+        )
+
+        assert found_proc
+
+    def test_proc_string_lookup_no_args(self):
+        found_proc = launch_testing.util.resolveProcesses(
+            self.proc_info,
+            process='good_proc',
+            cmd_args=launch_testing.util.NO_CMD_ARGS
```

Comment:
@pbaughman and if you do the above, this test will fail as the executable becomes `python path/to/script.py` and `NO_CMD_ARGS` no longer applies. Same thing happened to me on `test_io_handler_and_assertions.py`